Build error introduced for SYCL

Starting with commit 3727cca1 on the develop branch, files that include Kokkos_Core.hpp get a build error on SYCL:
```
In file included from .../kokkos/core/src/impl/Kokkos_hwloc.cpp:52:
In file included from .../kokkos/core/src/Kokkos_Core.hpp:51:
In file included from .../kokkos/core/src/Kokkos_Core_fwd.hpp:53:
In file included from .../kokkos/core/src/impl/Kokkos_Error.hpp:58:
.../kokkos-install/include/SYCL/Kokkos_SYCL_Abort.hpp:60:3: error: use of undeclared identifier '__assert_fail'
  __assert_fail(msg, empty, 0, empty);
```

Reproducer: clone develop branch and do
```
cmake -DCMAKE_CXX_COMPILER=dpcpp -DKokkos_ENABLE_SYCL=ON -DKokkos_ARCH_INTEL_XEHP=ON -DCMAKE_CXX_STANDARD=17 ../kokkos/example/build_cmake_in_tree
make -j
```

--- a/core/src/SYCL/Kokkos_SYCL_Abort.hpp
+++ b/core/src/SYCL/Kokkos_SYCL_Abort.hpp
@@ -47,6 +47,7 @@
 
 #include <Kokkos_Macros.hpp>
 #if defined(KOKKOS_ENABLE_SYCL)
+#include <CL/sycl.hpp>
 
 namespace Kokkos {
 namespace Impl {

--- a/example/build_cmake_in_tree/cmake_example.cpp
+++ b/example/build_cmake_in_tree/cmake_example.cpp
@@ -47,7 +47,7 @@
 
 int main(int argc, char* argv[]) {
   Kokkos::initialize(argc, argv);
-  Kokkos::DefaultExecutionSpace::print_configuration(std::cout);
+  Kokkos::DefaultExecutionSpace{}.print_configuration(std::cout);
 
   if (argc < 2) {
     fprintf(stderr, "Usage: %s [<kokkos_options>] <size>\n", argv[0]);

